pkg/handler/commit.go

```diff
@@ -441,3 +442,74 @@ func GetCommitStatuses(owner, repo, commitSHA string) ([]Status, error) {
 
 	return statusResp.Statuses, nil
 }
+
+func (h *commitHandler) GetCommitByAuthor(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	// Get request data from query params
+	request, errMessage := GetCommitByAuthorRequest(r)
+	if errMessage != "" {
+		http.Error(w, errMessage, http.StatusBadRequest)
+		return
+	}
+
+	// Construct the GitHub API URL for fetching commits by author
+	baseUrl := fmt.Sprintf("https://api.github.com/repos/%s/%s/commits?author=%s&per_page=100&page=", request.Owner, request.Repository, urlpkg.QueryEscape(request.Author))
+	method := "GET"
+
+	req, err := http.NewRequest(method, baseUrl, nil)
+	if err != nil {
+		http.Error(w, "Error generating new request: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Add authorization header if a personal access token is provided
+	if request.PersonalAccessToken != "" {
+		req.Header.Add("Authorization", "token "+request.PersonalAccessToken)
+	}
```

Comment:
I guess 2 more headers need to be set for Auth. Please use an existing method for this part - `AddhttpAuthRequestHeaders` or `AddAuthRequestHeaders`

Reply:
Using this is giving JSON unmarshalling error. Couldn't resolve it.